src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -120,6 +120,16 @@ namespace OpenMS
     bool has(String modification) const;
 
     /**
+       @brief Returns true if the modification is part of ModificationsDB
+    */
+    bool hasModification(const String & modification) const;
+
+    /**
+       @brief Add a new modification to ModificationsDB
+    */
+    void addModification(ResidueModification * new_mod);
+
+    /**
        @brief Returns the index of the modification in the mods_ vector; a unique name must be given
 
        @throw Exception::ElementNotFound if not exactly one matching modification was found
```

Comment:
can you add a comment that the user should probably call `hasModification` first to avoid an expensive `try - catch` ?